pkg/tabulator/tabstate.go

```diff
@@ -214,44 +232,44 @@ func Update(ctx context.Context, client gcs.ConditionalClient, mets *Metrics, co
 	for i := 0; i < concurrency; i++ {
 		go func() {
 			defer wg.Done()
-			for dashName := range dashboardNames {
+			for group := range channel {
 				lock.Lock()
-				start := active.Add(dashName)
+				start := active.Add(group.Name)
 				if !start {
```

Comment:
We can now have two go routines concurrently updating different parts of the same dashboard... 

What protects the one routine from updating the dashboard update done by another? The fact that the tab state is stored on a tab basis, unlike the summarizer which is stored on a dashboard basis (something to keep in mind in case a future refactor violates this hidden design dependency).